### PR TITLE
Fix crash when walking sideways

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/16045/crash-on-strafing
The bug described above showed up after this PR: https://github.com/highfidelity/hifi/pull/13304
The bug occurred because there was some code being used that was never used before, causing some NaNs to show up that were previously ignored.
The solution, given in this PR, is to get rid of the NaNs.
This is a proper fix to take the place of this PR: https://github.com/highfidelity/hifi/pull/13420

## TEST PLAN
* In master, verify program crashes when strafing with the avatar for a while.
* In this PR, verify crash goes away.
* In this PR, verify that avatar forward/sideways/diagonal/backwards movement is still as expected.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10749/HighFidelity-Beta-PR13425-45fdf3d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10749/HighFidelity-Beta-PR13425-45fdf3d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10749/HighFidelity-Beta-Interface-PR13425-45fdf3d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10749/HighFidelity-Beta-PR13425-45fdf3d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10749/HighFidelity-Beta-Interface-PR13425-45fdf3d.exe

File: interface/src/avatar/MyCharacterController.cpp [hunk 1]
```diff
@@ -108,12 +108,15 @@ bool MyCharacterController::testRayShotgun(const glm::vec3& position, const glm:
     btScalar lengthAxis = axis.length();
     if (lengthAxis > FLT_EPSILON) {
         // we're walking sideways
-        btScalar angle = acosf(lengthAxis / adjustedDirection.length());
-        if (rayDirection.dot(forward) < 0.0f) {
-            angle = PI - angle;
+        btScalar cosAngle = lengthAxis / adjustedDirection.length();
+        if (cosAngle < 1) {
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyCharacterController.cpp]: As per the coding standard (which seems used to be located here but is currently lost) we want all float literals to have at least one decimal point and to end with `f`, so you need to change `1` to be `1.0f`.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10759/HighFidelity-Beta-PR13425-a8b2da1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10759/HighFidelity-Beta-PR13425-a8b2da1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10759/HighFidelity-Beta-Interface-PR13425-a8b2da1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10759/HighFidelity-Beta-PR13425-a8b2da1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10759/HighFidelity-Beta-Interface-PR13425-a8b2da1.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Interface does not crash.

**Pull request merged**